cpp/unit/unit.h

```diff
@@ -43,9 +43,9 @@ class Unit : public log::LogSource {
 	const id_t id;
 
 	/**
-	 * producer used to create this object
+	 * type which was used to create this object
```

Comment:
Yes, calling type.initialise(unit, player) on an already typed unit will reset the unit to the new type